lab_3_generate_by_ngrams/main.py

```diff
@@ -388,6 +562,38 @@ def run(self, prompt: str, seq_len: int) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not (isinstance(prompt, str) and prompt
+                and isinstance(seq_len, int) and seq_len):
+            return None
+        encoded_prompt = self._text_processor.encode(prompt)
+        if not encoded_prompt:
+            return None
+        candidates = {encoded_prompt: 0.0}
+        for i in range(seq_len):
+            new_candidates = dict(candidates)
+            for sequence in candidates:
+                next_tokens = self._get_next_token(sequence)
+                if not next_tokens:
+                    return None
+                continued_sequence = (
+                    self.beam_searcher.continue_sequence(sequence, next_tokens, new_candidates)
+                )
+                if not continued_sequence:
```

Comment:
Consider breaking the cycle instead of duplicating code

Reply:
Done